### ISSUE TITLE
dynamic references to kubectl in kraken.services

### ISSUE THREAD

**Issue by coffeepac** [samsung-cnct/k2#247]
as part of #222 we are going to be supporting multiple versions of kubernetes.  as such we will need multiple versions of kubectl.  the exact version of kubectl to execute for a given command should be config driven exactly as the helm commands are in #236, e.g.

```
/opt/cnct/kubernetes/{{ kubernetes_minor_version }}/bin/kubectl
```

**Comment by coffeepac** [comment]
blocked by #248

**Comment by guineveresaenger** [comment]
As per k2recon, we will also have multiple kubernetes minor versions floating about:
https://github.com/samsung-cnct/k2/pull/349 
It occurs to me that this will need to be migrated to the new format as well. Does this warrant a new ticket?

**Comment by coffeepac** [comment]
yes, this will need to reference the per-cluster fact

**Comment by guineveresaenger** [comment]
I made some headway on this on master, and am now waitinf for k2recon to be either merged or ready to run cluster up so I can make it per cluster.
branch for reference: https://github.com/guineveresaenger/k2/tree/kubectlfix

**Comment by guineveresaenger** [comment]
PR here: #363

**Comment by guineveresaenger** [comment]
Concern: Kubectl gets called in a couple other places as well:
- kraken.fabric.flannel (deploy-canal)
- kraken.readiness  (bootstrap and generate admin-kubeconfig)

Comments from @mikeln at #363 as well

**Comment by coffeepac** [comment]
@guineveresaenger can you expand this to cover those other two roles?  when I wrote this ticket I was unaware of those uses.

**Comment by guineveresaenger** [comment]
@coffeepac absolutely! back to the drawing board

**Comment by coffeepac** [comment]
PR was merged.
